static/app/views/eventsV2/index.tsx

```diff
@@ -7,10 +10,30 @@ import withOrganization from 'sentry/utils/withOrganization';
 
 type Props = {
   children: React.ReactChildren;
+  location: any;
   organization: Organization;
 };
 
-function DiscoverContainer({organization, children}: Props) {
+function DiscoverContainer({organization, children, location}: Props) {
+  const prevLocationPathname = useRef('');
+
+  useEffect(
+    // when new discover page loads, query is pristine
+    function () {
+      if (location.pathname !== prevLocationPathname.current) {
+        prevLocationPathname.current = location.pathname;
+        browserHistory.push({
+          pathname: location.pathname,
+          query: {
+            ...location.query,
+            isDirty: false,
+          },
+        });
+      }
+    },
+    [location]
```

Comment:
maybe double check these top level useEffects aren't causing any rerenders?

Reply:
it seeeeems okay, no unnecessary rerenders noticed